### PR TITLE
Openscenegraph/3.6.5 opengl profile

### PR DESCRIPTION
**Pull request by Makamitsu**
It would be convenient if the "opengl_profile" attribute was exposed in the conan recipe. It currently works like a charm on my env.

I didn't have the time to take care of the merge of this PR last year, that's why I'm reopening this PR.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by KrisNumber24 — DISMISSED** [review]
Works with my dev environment.

**Review by spb-fdd — DISMISSED** [review]
Tested successfully on my environment.

**Review by kaloskagatos — APPROVED** [review]
Works for me thank you ! 🐱‍👤

**Comment by prince-chrismc** [comment]
You should consider making this recipe ready for v2 :)

**Review by uilianries — APPROVED** [review]
It could be improved for Conan v2, but so far, it's good enough for a new option PR.

File: recipes/openscenegraph/all/conanfile.py [hunk 1]
```diff
@@ -186,6 +194,8 @@ def _configured_cmake(self):
 
         cmake.definitions["OSG_TEXT_USE_FONTCONFIG"] = self.options.use_fontconfig
 
+        cmake.definitions["OPENGL_PROFILE"] = str(self.options.get_safe("opengl_profile", "gl2")).upper()
```
Comment (prince-chrismc) [inline recipes/openscenegraph/all/conanfile.py:197]: This option is not deleted, lets try to keep it simple 

```suggestion
        cmake.definitions["OPENGL_PROFILE"] = str(self.options.opengl_profile).upper()
```

File: recipes/openscenegraph/all/conanfile.py [hunk 2]
```diff
@@ -1,9 +1,15 @@
-from conans import CMake, ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.tools.files import get, rmdir, rm, patch
```
Comment (prince-chrismc) [inline recipes/openscenegraph/all/conanfile.py:2]:
```suggestion
from conan.tools.files import get, rmdir, rm, patch, apply_conandata_patches
```

File: recipes/openscenegraph/all/conanfile.py [hunk 3]
```diff
@@ -149,12 +157,12 @@ def requirements(self):
...
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
+        get(self, **self.conan_data["sources"][self.version],
                   strip_root=True, destination=self._source_subfolder)
 
     def _patch_sources(self):
-        for patch in self.conan_data["patches"].get(self.version, []):
-            tools.patch(**patch)
+        for p in self.conan_data["patches"].get(self.version, []):
+            patch(self, **p)
```
Comment (prince-chrismc) [inline recipes/openscenegraph/all/conanfile.py:165]:
```suggestion
        apply_conandata_patches(self)
```

**Review by prince-chrismc — COMMENTED** [review]
This recipe needs a lot of improvements :) you should consider following it up if you want to keep it working

File: recipes/openscenegraph/all/conanfile.py [hunk 4]
```diff
@@ -1,9 +1,15 @@
-from conans import CMake, ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.tools.files import get, rmdir, rm, patch, apply_conandata_patches
```
Comment (prince-chrismc) [inline recipes/openscenegraph/all/conanfile.py:2]:
```suggestion
from conan.tools.files import get, rmdir, rm, apply_conandata_patches
```

File: recipes/openscenegraph/all/conanfile.py [hunk 5]
```diff
@@ -149,12 +157,11 @@ def requirements(self):
             self.requires("zlib/1.2.12")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
+        get(self, **self.conan_data["sources"][self.version],
                   strip_root=True, destination=self._source_subfolder)
```
Comment (prince-chrismc) [inline recipes/openscenegraph/all/conanfile.py:161]:
```suggestion
        get(self, **self.conan_data["sources"][self.version], strip_root=True)
```

**Comment by prince-chrismc** [comment]
I dont know why bzip was missing but hopefully I fixed the other problems

File: recipes/openscenegraph/all/conanfile.py [hunk 6]
```diff
@@ -138,23 +146,21 @@ def requirements(self):
...
         if self.options.with_tiff:
             self.requires("libtiff/4.3.0")
         if self.options.with_zlib:
-            self.requires("zlib/1.2.12")
+            self.requires("zlib/1.2.13")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
```
Comment (prince-chrismc) [inline recipes/openscenegraph/all/conanfile.py:160]:
```suggestion
        get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```

**Review by prince-chrismc — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**